### ISSUE TITLE
Wrong LASTTICK

### ISSUE THREAD

**Issue by bensmrs** [senx/warp10-platform#458]
Hi! me again :)

I found a very strange bug that I'm unable to reproduce by hand, so I've put the data needed to reproduce it in a gist: https://gist.github.com/bensmrs/c064434943d418f6533e08cfb1bba20f
Basically, LASTTICK doesn't return the actual last tick.

Can you check that?

My example should return this:
```
[TOP] 1560203940000000
2     1560266220000000
```

**Comment by bensmrs** [comment]
As a side effect, every MAP I try to apply to that GTS stops at the tick returned by LASTTICK, not the true one, so quite embarrassing…

**Comment by ftence** [comment]
Hi!

This is because your GTS was bucketized and then some values were added. LASTTICK returns the newest tick or the last bucket if the GTS is bucketized. Here is a simple example showing the same behavior.

```
NEWGTS
0 NaN NaN NaN 1 ADDVALUE
10 NaN NaN NaN 1 ADDVALUE
20 NaN NaN NaN 1 ADDVALUE

[ SWAP bucketizer.last 20 10 0 ] BUCKETIZE
0 GET
30 NaN NaN NaN 1 ADDVALUE

SORT DUP
TICKLIST DUP SIZE 1 - GET // Get the last tick using TICKLIST
SWAP LASTTICK // Get the last tick using LASTTICK
```

The solution is to use `UNBUCKETIZE` on your GTS.

**Comment by bensmrs** [comment]
Awesome! Is this behaviour documented anywhere?

But there is something that disturbs me a bit: I did not add any value to the GTS…
I used a FETCH to get several GTSes, bucketized them, and calculated the ratio between two sums: the sum of the GTSes matching a specific label and the total sum. The sums are calculated with `reducer.sum` using `REDUCE` and the ratio is calculated with `op.div` using `APPLY`. The metadata saying « I'm bucketized » is apparently kept, but the « last known bucketized tick » is the wrong one then.

**Comment by ftence** [comment]
The behavior of LASTTICK concerning bucketized GTSs is not documented. I updated it and it should be online in a few days.

You indeed spotted something weird, most probably a bug, thanks for that. Combining GTSs with different last buckets with REDUCE (and maybe APPLY) results in GTSs with what seems a wrong last bucket. We're investigating that.

For now, I would recommend setting the same last bucket on all your GTSs (don't use the magic `0` parameter). You can use the LASTTICK function on the list returned by FETCH to use that value.

**Comment by hbs** [comment]
Thanks for the catch, there was indeed an issue with the REDUCE, it is fixed in 250a102.
